lib/ddtrace/context.rb

```diff
@@ -88,17 +88,20 @@ def current_root_span
       end
     end
 
-    # Add a span to the context trace list, keeping it as the last active span.
-    def add_span(span)
+    def add_span(operation)
       @mutex.synchronize do
+        # Make the span follow the current span, unless there isn't one.
+        operation.trace_id = @parent_trace_id unless @parent_trace_id.nil?
+        operation.parent_id = @parent_span_id unless @parent_span_id.nil?
```

Comment:
I noticed this was moved from `#start_span` to here, so it's not a new concern, but isn't it dangerous to set `parent_id=` without also setting `parent=`?

Reply:
`SpanOperation#new` and `SpanOperation#parent=` do this now, instead of `Tracer#start_span`. I think this specific behavior happens here because of distributed tracing.

Need to look at this more carefully.

Reply:
> I think this specific behavior happens here because of distributed tracing.

I can see distributed tracing as a valid case for having the `parent_id` but not a `parent` object, which unfortunately makes the span API inconsistent.

It looks to me that we should deprecate the `attr_reader :parent` API in the future, as it doesn't work on all cases and be create misleading scenarios where `span.parent == nil`, but `span.parent_id != nil` because the span is a distributed span.